main.go

```diff
@@ -116,37 +144,85 @@ func main() {
 	m := &Miner{
 		config:         config,
 		engine:         blake3Engine,
-		sliceClients:   connectToSlice(config),
 		header:         types.EmptyHeader(),
-		updateCh:      make(chan *types.Header, resultQueueSize),
+		updateCh:       make(chan *types.Header, resultQueueSize),
 		resultCh:       make(chan *types.Header, resultQueueSize),
-		previousNumber: [common.HierarchyDepth]uint64{0,0,0},
+		previousNumber: [common.HierarchyDepth]uint64{0, 0, 0},
 	}
 	log.Println("Starting Quai cpu miner in location ", config.Location)
-	m.fetchPendingHeader()
-	go m.subscribePendingHeader()
-	go m.resultLoop()
+	if config.Proxy {
+		m.proxyClient = connectToProxy(config)
+		go m.fetchPendingHeaderProxy()
+		go m.startProxyListener()
+		go m.subscribeProxy()
+	} else {
+		m.sliceClients = connectToSlice(config)
+		go m.fetchPendingHeaderNode()
+		// No separate call needed to start listeners.
```

Comment:
no need for this comment right

Reply:
Discussed irl, understood that this comment is there to provide symmetry with the proxy/non-proxy modes